src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -178,4 +211,526 @@ export class NetworkAreaDiagramViewer {
+        );
+    }
+
+    private updateGraph(event: Event) {
+        const offset = this.getMousePosition(event as MouseEvent);
+        this.moveNode(offset);
+        this.moveNodeText(offset);
+        this.moveEdges();
+    }
+
+    private moveNode(offset: Point) {
+        //this.transform = DiagramUtils.getTransform(this.selectedElement);
+        //this.transform?.setTranslate(offset.x, offset.y);
+        this.selectedElement?.setAttribute(
+            'transform',
+            'translate(' + offset.x.toFixed(2) + ',' + offset.y.toFixed(2) + ')'
+        );
+    }
+
+    private moveNodeText(offset: Point) {
+        const translation = new Point(
+            offset.x - this.initialPosition.x,
+            offset.y - this.initialPosition.y
+        );
+        // move node text
+        this.moveTextNode(this.selectedElement?.id + '-textnode', translation);
+        // move edge connecting node and node text
+        this.moveTextNode(this.selectedElement?.id + '-textedge', translation);
+    }
+
+    private moveTextNode(textNodeId: string, translation: Point) {
+        const textNode: SVGGraphicsElement | null =
+            this.container.querySelector("[id='" + textNodeId + "']");
+        if (textNode) {
+            const transform = DiagramUtils.getTransform(textNode);
+            const totalTranslation = new Point(
+                (transform?.matrix.e ?? 0) + translation.x,
+                (transform?.matrix.f ?? 0) + translation.y
+            );
+            //transform?.setTranslate(totalTranslation.x, totalTranslation.y);
+            textNode?.setAttribute(
+                'transform',
+                'translate(' +
+                    totalTranslation.x.toFixed(2) +
+                    ',' +
+                    totalTranslation.y.toFixed(2) +
+                    ')'
+            );
+        }
+    }
+
+    private moveEdges() {
+        // get edges connected to the the node we are moving
+        const edges: NodeListOf<SVGGraphicsElement> =
+            this.container.querySelectorAll(
+                'nad\\:edge[node1="' +
+                    (this.selectedElement?.id ?? -1) +
+                    '"], nad\\:edge[node2="' +
+                    (this.selectedElement?.id ?? -1) +
+                    '"]'
+            );
+        // group edges, to have multibranches - branches connecting the same nodes - together
+        const groupedEdges: Map<string, SVGGraphicsElement[]> = new Map<
+            string,
+            SVGGraphicsElement[]
+        >();
+        edges.forEach((edge) => {
+            const node1 = edge.getAttribute('node1') ?? '-1';
+            const node2 = edge.getAttribute('node2') ?? '-1';
+            let edgeGroup: SVGGraphicsElement[] = [];
+            if (groupedEdges.has(node1.concat(node2))) {
+                edgeGroup = groupedEdges.get(node1.concat(node2)) ?? [];
+            }
+            edgeGroup.push(edge);
+            groupedEdges.set(node1.concat(node2), edgeGroup);
```

Comment:
You should introduce a variable for `node1.concat(node2)` (or for what will replace it)